Classes/Seo/ImagesXmlSitemapDataProvider.php

```diff
@@ -34,57 +38,46 @@
  */
 class ImagesXmlSitemapDataProvider extends AbstractXmlSitemapDataProvider
 {
-    /**
-     * @var ImageFileReferenceRepository
-     */
-    private ImageFileReferenceRepository $imageFileReferenceRepository;
+    private readonly ImageFileReferenceRepository $imageFileReferenceRepository;
+    private readonly PageRepository $pageRepository;
+    private readonly SiteFinder $siteFinder;
+    private readonly LinkFactory $linkFactory;
 
     /**
-     * @var UriBuilder
-     */
-    private UriBuilder $uriBuilder;
-
-    /**
-     * Constructor.
-     *
-     * @param ServerRequestInterface     $request
-     * @param string                     $key
-     * @param array                      $config
-     * @param ContentObjectRenderer|null $cObj
-     *
-     * @throws Exception
      * @throws InvalidQueryException
-     * @throws \Doctrine\DBAL\Driver\Exception
+     * @throws MissingConfigurationException
+     * @throws Exception
      */
     public function __construct(
         ServerRequestInterface $request,
         string $key,
         array $config = [],
-        ContentObjectRenderer $cObj = null
+        ?ContentObjectRenderer $cObj = null
     ) {
         parent::__construct($request, $key, $config, $cObj);
 
-        $objectManager = GeneralUtility::makeInstance(ObjectManager::class);
+        $connectionPool = GeneralUtility::makeInstance(ConnectionPool::class);
+        $context        = GeneralUtility::makeInstance(Context::class);
 
         $this->imageFileReferenceRepository
-            = $objectManager->get(ImageFileReferenceRepository::class);
-        $this->uriBuilder
-            = $objectManager->get(UriBuilder::class);
+            = GeneralUtility::makeInstance(ImageFileReferenceRepository::class, $connectionPool, $context);
+        $this->pageRepository = GeneralUtility::makeInstance(PageRepository::class);
+        $this->siteFinder = GeneralUtility::makeInstance(SiteFinder::class);
+        $this->linkFactory = GeneralUtility::makeInstance(LinkFactory::class);
```

Comment:
not sure, but shouldn't this be handled by propper DI?

as described here: https://docs.typo3.org/m/typo3/reference-coreapi/main/en-us/ApiOverview/DependencyInjection/Index.html

@aseemann what do you think?